### PR TITLE
selector garbage collection, use `selection` feature name, cleanup event handlers

### PR DESCRIPTION
**Pull request by clewis7**

### CODE REVIEW

**Comment by kushalkolar** [comment]
let's also throw in #211

File: fastplotlib/graphics/features/_base.py [hunk 1]
```diff
@@ -254,6 +259,9 @@ def cleanup_array_slice(key: np.ndarray, upper_bound) -> np.ndarray:
     if key.dtype == bool:
         key = np.nonzero(key)[0]
 
+    if key.size < 1:
```
Comment (kushalkolar) [inline fastplotlib/graphics/features/_base.py:262]: I forgot what this fixed :smile:

File: fastplotlib/graphics/selectors/_linear_region.py [hunk 2]
```diff
@@ -11,6 +11,9 @@
 
 
 class LinearBoundsFeature(GraphicFeature):
+    feature_events = (
+        "data",
```
Comment (kushalkolar) [inline fastplotlib/graphics/selectors/_linear_region.py:15]: this is supposed to be bounds, but as for #211 let's rename it to selection

File: fastplotlib/graphics/features/_base.py (hunk 1 above)
Comment (kushalkolar) [reply fastplotlib/graphics/features/_base.py:262]: @clewis7 can you jog my memory? Is this when you do something like `line_graphic.colors[xs > 0] = "w"`, but `xs > 0` is an empty array?

Comment (clewis7) [reply fastplotlib/graphics/features/_base.py:262]: yes, bc we kept getting an error when trying to set the colors and a behavior had not a occurred so the array was all zeroes, would never give True

**Comment by kushalkolar** [comment]
this got included in #207

**Pull request closed without merging**